src/index.js

```diff
@@ -11,214 +11,224 @@ const Bucket = core.getInput('bucket');
-
-    const pdf = await page.pdf({
-        format: 'A4',
-        printBackground: true,
-        displayHeaderFooter: true,
-        footerTemplate,
-        margin: {
-            top: '1.2cm',
-            bottom: '2.3cm',
-            right: '1.5cm',
-            left: '1.5cm',
-        },
-    });
-    await browser.close();
-
-    const mergedPdf = await appendPDF(basePDF, pdf);
-    console.log('Done, starting the uploads...');
-
-    /**
-     * uploading PDF to S3
-     */
-    await uploadFile(mergedPdf, `versions/${pdfName}`);
-    await uploadFile(mergedPdf, 'latest/Limbic Access - Instructions for Use (IFU).pdf');
+  const mapped = await mapAllFiles(pdfName);
+  if (!mapped) return;
+
+  /**
+   * downloading base PDF and templates
+   */
+  const basePDF = await getAsset(basePdf);
+  const bodyTemplate = await getTemplateHtml('pdf_v2.html');
+  const footerTemplate = await getTemplateHtml('footer.html');
+
+  /**
+   * creating the template from html using handlebars
+   */
+  const template = await hb.compile(bodyTemplate, {strict: true});
+  const html = template({
+    version: await getVersion(),
+    shortVersion: await getVersion('short'),
+    releaseDate: await getReleaseDate(),
+    previousFiles: mapped,
+    manufacturer: readFileSync('/assets/manufacturer.png').toString('base64'),
+    dateManufacturer: readFileSync('/assets/dateManufacturer.png').toString(
+      'base64'
+    ),
+    ref: readFileSync('/assets/ref.png').toString('base64'),
+    lot: readFileSync('/assets/lot.png').toString('base64'),
+    udi: readFileSync('/assets/udi.png').toString('base64'),
+    ukca: readFileSync('/assets/ukca.png').toString('base64'),
+    caution: readFileSync('/assets/caution.png').toString('base64'),
+    eifu: readFileSync('/assets/eifu.png').toString('base64'),
+    logo: readFileSync('/assets/logo.png').toString('base64'),
+    md: readFileSync('/assets/md.png').toString('base64'),
+  });
+
+  /**
+   * launching puppeteer to generate the version PDF
+   */
+  console.log(`Starting PDF generation with ${pdfName} name`);
+  const browser = await puppeteer.launch({
+    args: ['--no-sandbox', '--disable-setuid-sandbox'],
+    headless: true,
+    executablePath: 'google-chrome-stable',
+  });
+  const page = await browser.newPage();
+  await page.setContent(html, {
+    waitUntil: ['load', 'domcontentloaded', 'networkidle0'],
+  });
+  await page.addStyleTag({path: '/styles/pdf.css'});
+
+  const pdf = await page.pdf({
+    format: 'A4',
+    printBackground: true,
+    displayHeaderFooter: true,
+    footerTemplate,
+    margin: {
+      top: '1.2cm',
+      bottom: '2.3cm',
+      right: '1.5cm',
+      left: '1.5cm',
+    },
+  });
+  await browser.close();
+
+  const mergedPdf = await appendPDF(basePDF, pdf);
+  console.log('Done, starting the uploads...');
+
+  /**
+   * uploading PDF to S3
+   */
+  await uploadFile(mergedPdf, `versions/${pdfName}`);
+  await uploadFile(mergedPdf, outputPdfKey);
```

Comment:
I agree with setting the pdf name from the github action but I think we should keep the rest.
so` latest/${latestPdfName}` or something. I don't think it's clear `outputPdfKey` has to have the path and that's not what is happening the line above either where we kept the versions/${pdfName}. 

Unless for care this setup is not possible because it's necessary to have the "latest" pdf in a completely different place